### ISSUE TITLE
multiple are not send to POST when submit

### ISSUE THREAD

**Issue by mckaygerhard** [Symphony9/pickathing#2]
i have a combo box select with the tag multiple:

```
<td>Entidad Destino:</td><td><select name="cod_entidad"   multiple="multiple" id="cod_entidad_id">
<option value="000">(0) ALM Almacen (NA)</option>
<option value="034">(34) RCS Los Ruices ()</option>
<option value="036">(36) SBL Deposito Sabal ()</option>
</select><script src="http://127.0.0.1/~systemas/public_html/pickathing.js"></script><script>var selectcod_entidad = new Pickathing('cod_entidad_id', true);</script>
```

when i click the submit button the post parameters does not register the input "cod_entidad", i mean cod_entidad value are not parsed never

there's any special that i forget to do, or well how can i contribute to solve it?

**Comment by mckaygerhard** [comment]
i got the post with a [] but if use the pickathing get only 
`Array ( [0] =&gt `
if i disable the pickathing got complete array post multiselect:
`<pre>Array ( [0] =&gt; 000 [1] =&gt; 034 [2] =&gt; 036 ) </pre>`
please, some help

**Comment by VojtechKlos** [comment]
@mckaygerhard Hey. If I understand it correctly you do not get any values sent with the form after submit? The values in multiple selects are very hard to implement because in js you do not get them as an array or anything similar. What we do to overcome this issue is to create a dummy text input which we then fill with the values from Pickathing.value. So for example:

`
// HTML
<select multiple="true" id="cod_entidad_id">
    <option value="1">option 1</option>
    <option value="2">option 2</option>
</select>
<input type="text" id="actual-value">

// JS
var selectcod_entidad = new Pickathing('cod_entidad_id', true);
var input = document.getElementById('actual-value');
selectcod_entidad.onChange = function() {
    input.value = JSON.stringify(Object.keys(selectcod_entidad));
}
`

after that the value of the input will be stringified version of the array made of values of the multiple select.


Or you can try to send the form via ajax with json data and set the valu explicitly to `selectcod_mentidad.value`.

Let me know if my answer helps ;-)

**Comment by mckaygerhard** [comment]
thanks for u'r quick answer and info, previously i get a "value", but "cutted", i try with others libraries like Mobius1/Selectr and works, sorry for my english and lack of the complete environment, i implemented u'r code in the codeigniter own framework for my projects..

i'm not expert, i'm electricits and i hate any js/ajax/jquery/json code, i try u'r suggestion but its not dessired result, due i lack of explication of why i got array, firts let me know to u that i generate correctly the normal select combobox and get the value in normal select box..

i posted as array due in codeigniter i echo-ed as array, but in any case, in core php i not get it either, u'r anwers helps, but need a extra input, and i do not want to complicate the code, i mean, in the framework i do only:

`echo form_multiselect('cod_entidad[]', $list_entidad_destino,null, 'id="selectid2"')`

and i override/edited the code of the form_helper of codeigniter and generates correctly (as u'r documentation said) the html+js code as:

```
<select name="cod_entidad"  id="selectid2">
<option value="000">(0) ALM Almacen (NA)</option>
<option value="034">(34) RCS Los Ruices ()</option>
<option value="035">(35) PAN Colon Panama ()</option>
</select><script src="http://127.0.0.1/~systemas/appweb/scripts/pickathing.js"></script><script>var selectcod_almacen = new Pickathing('cod_almacen', true);</script>
```

with $list_entidad as an array with the par of calue/text of the select combobox, so if i need a extra input i'll need to put more complicated logic ar the "form_multiselect" fuction in the helper, the idea its to avoid any complicated logic like ajax and extra input files, that's why i use u'r javascript implementation

currently (due i liked more) i used u'r javascript code only for normal select comboboxes, and for multiselect i use the Mobius1/Selectr and put a config option to choose what implementation use..

i try u'r suggestion but its not the dessire result, i think due i not explain why i got an array of the select multi combobox..

**Comment by mckaygerhard** [comment]
edited sorry corrected the input and id names cod_entidad

ad thanks for u'r excelent javascript library, in comparison of Mobius1/Selectr, i implemnet fast and easy u'r pickathing select, around 1.23 minutes of code in helper and view and ready results !

**Comment by VojtechKlos** [comment]
Glad to be of help.

The generated HTML looks fine. But in case of mulltiple selects my library does not manipulate the original multiselect in any way. That is why you are getting an empty array because right now this functionality is not implemented. Note that this is a case only with multiselects - normal selects work just fine. I am dedicated to implement this feature in a future release but for now you have to use the workaround I posted before.

This project started as I needed a vanilla js library that does not use jQuery or anything else for a project that I am currently working on so I implemented only functionality I needed so far. But your suggestion seems logical so I will dedicate some time to implement it. No promises on time though ;-)

Thanks for your kind words! Cheers!

**Comment by mckaygerhard** [comment]
okey, good to know and i want to thanks for u'r works, was very very easy to implement, ...

... please keep the work and i wait for the multiselect feature (i need desesperatelly in pickathing, i dont like the setectr), wheantime i'l make and article about usage in codeigniter, so i'll waith for u'r commits in this repository

**Comment by mckaygerhard** [comment]
i forgotten, it returns but inly the firts value selected! in multiselect cases

**Comment by VojtechKlos** [comment]
Will keep you updated on my work. I think maybe next week I can implement this feature.

Cheers and thanks again ;-)

**Comment by VojtechKlos** [comment]
@mckaygerhard Hello again. Just pushed some updates which should hopefully address your issue. Just download the new version(0.8.0) from npm or download the master branch from github.

Let me know if it works ;-)
Cheers!

**Comment by mckaygerhard** [comment]
YEAH! its working, now i send a new issue, (yep, but a issue very little maybe)

may thanks, i like more the pickathing closes wiith 2058f95
